crates/fuel-core/src/state/rocks_db.rs

```diff
@@ -450,28 +447,26 @@ where
     where
         T: ExtractItem,
     {
-        let maybe_next_item = next_prefix(prefix.to_vec())
-            .and_then(|next_prefix| {
-                self.iter_store(
-                    column,
-                    Some(next_prefix.as_slice()),
-                    None,
-                    IterDirection::Forward,
-                )
-                .next()
+        let reverse_iterator = next_prefix(prefix.to_vec()).map(|next_prefix| {
+            let mut opts = self.read_options();
+            opts.set_total_order_seek(true);
+            self.iterator::<T>(
+                column,
+                opts,
+                IteratorMode::From(next_prefix.as_slice(), rocksdb::Direction::Reverse),
+            )
+            .skip_while(move |item| {
+                if let Ok(item) = item {
+                    T::starts_with(item, next_prefix.as_slice())
+                } else {
+                    false
+                }
```

Comment:
No from what I tested it doesn't work like that as the rocksDB function use by our iterator is `SeekForPrev` we will not go after `2` and so if two doesn't exists it will take the key that is the nearest but lower than `2`. In your case it's 1. I will add tests and you will see it from tests.

Reply:
Can you create a followup issue for the optimization? And add a TODO in the code?

Reply:
https://github.com/FuelLabs/fuel-core/issues/2405